File: backend/packages/contracts/src/contracts/seeking.py
from __future__ import annotations

from pydantic import BaseModel, ConfigDict, Field


class ScanSummary(BaseModel):
    model_config = ConfigDict(extra="forbid", populate_by_name=True)

    total_scanned: int = Field(alias="totalScanned")
    high_matches: int = Field(alias="highMatches")
    medium_matches: int = Field(alias="mediumMatches")
    period_label: str = Field(alias="periodLabel")


class SkillMatch(BaseModel):
    model_config = ConfigDict(extra="forbid")

    skill: str
    matched: bool
    level: str


class HighMatchItem(BaseModel):
    model_config = ConfigDict(extra="forbid", populate_by_name=True)

    match_id: str = Field(alias="matchId")
    job_id: str = Field(alias="jobId")
    job_title: str = Field(alias="jobTitle")
    company_name: str = Field(alias="companyName")
    location: str
    work_mode: str = Field(alias="workMode")
    score: float
    skill_matches: list[SkillMatch] = Field(alias="skillMatches")
    ai_assessment: str = Field(alias="aiAssessment")


class PreChatItem(BaseModel):
    model_config = ConfigDict(extra="forbid", populate_by_name=True)

    inbox_item_id: str = Field(alias="inboxItemId")
    company_name: str = Field(alias="companyName")
    job_title: str = Field(alias="jobTitle")
    summary: str
    generated_at: str = Field(alias="generatedAt")


class InboundInterestItem(BaseModel):
    model_config = ConfigDict(extra="forbid", populate_by_name=True)

    match_id: str = Field(alias="matchId")
    company_name: str = Field(alias="companyName")
    reason: str
    score: float
    job_id: str = Field(alias="jobId")


class SeekingReportData(BaseModel):
    model_config = ConfigDict(extra="forbid", populate_by_name=True)

    scan_summary: ScanSummary = Field(alias="scanSummary")
    high_matches: list[HighMatchItem] = Field(alias="highMatches")
    pre_chat_activity: list[PreChatItem] = Field(alias="preChatActivity")
    inbound_interest: list[InboundInterestItem] = Field(alias="inboundInterest")
    generated_at: str = Field(alias="generatedAt")


class SeekingResponse(BaseModel):
    data: SeekingReportData | None


class ResumeGenerateRequest(BaseModel):
    model_config = ConfigDict(extra="forbid", populate_by_name=True)

    talent_id: str = Field(alias="talentId", min_length=1)
    job_id: str = Field(alias="jobId", min_length=1)


class TailoredResumePayload(BaseModel):
    model_config = ConfigDict(extra="forbid", populate_by_name=True)

    markdown: str
    talent_name: str = Field(alias="talentName")
    job_title: str = Field(alias="jobTitle")
    company_name: str = Field(alias="companyName")


class TailoredResumeResponse(BaseModel):
    data: TailoredResumePayload

File: src/lib/api/seeking.ts
import { apiFetch } from '@/lib/api/client';

export interface ScanSummary {
  totalScanned: number;
  highMatches: number;
  mediumMatches: number;
  periodLabel: string;
}

export interface HighMatchItem {
  matchId: string;
  jobId: string;
  jobTitle: string;
  companyName: string;
  location: string;
  workMode: string;
  score: number;
  skillMatches: Array<{
    skill: string;
    matched: boolean;
    level: string;
  }>;
  aiAssessment: string;
}

export interface PreChatItem {
  inboxItemId: string;
  companyName: string;
  jobTitle: string;
  summary: string;
  generatedAt: string;
}

export interface InboundInterestItem {
  matchId: string;
  companyName: string;
  reason: string;
  score: number;
  jobId: string;
}

export interface SeekingReportData {
  scanSummary: ScanSummary;
  highMatches: HighMatchItem[];
  preChatActivity: PreChatItem[];
  inboundInterest: InboundInterestItem[];
  generatedAt: string;
}

export async function getLatestReportByTalentId(
  _talentId?: string
): Promise<SeekingReportData | null> {
  const response = await apiFetch<{ data: SeekingReportData | null }>('/api/v1/seeking');
  return response.data;
}

export async function getLatestReportByUserId(
  _userId?: string
): Promise<SeekingReportData | null> {
  const response = await apiFetch<{ data: SeekingReportData | null }>('/api/v1/seeking');
  return response.data;
}

export async function upsertReport(
  talentId: string,
  reportData: SeekingReportData
): Promise<void> {
  throw new Error(`Frontend seeking writes are no longer supported for ${talentId}:${reportData.generatedAt}`);
}
